py/desispec/io/spectra.py

```diff
@@ -186,7 +186,25 @@ def write_spectra(outfile, spec, units=None):
     return outfile
 
 
-def read_spectra(infile, single=False):
+def read_spectra(
+    infile,
+    single=False,
+    targetids=None,
+    skip_hdu={
+        "FIBERMAP": False,
+        "EXP_FIBERMAP": False,
+        "SCORES": False,
+        "EXTRA_CATALOG": False,
+        "MASK": False,
+        "RESOLUTION": False,
+    },
```

Comment:
As currently implemented, this has the nice feature of self-documenting what the options are, but has the downside of always having to specify all of them even if you only want to skip one of them.  e.g. if the user wants to skip the RESOLUTION HDU but keep the others, they still have to specify `skip_hdu['EXTRA_CATALOG']: False` even if the input file doesn't even have an EXTRA_CATALOG HDU, because otherwise the line `if not skip_hdu["EXTRA_CATALOG"]` would crash.

I suggest making this a list/set `skip_hdus` where the user provides the list/set of HDUs to skip, without having to explicitly also include the ones not to skip.  Downstream code would then become like
```python
if name == 'FIBERMAP' and 'FIBERMAP' not in skip_hdus:
    ...
```
In this case the docstring should also be updated to list which HDUs can be skipped since the default argument doesn't self-document that.

Alternatively, you could keep this as a dict but change the downstream code to be something like
```python
if (name == 'FIBERMAP') and ('FIBERMAP' in skip_hdus) and skip_hdus['FIBERMAP']:
    ...
```
which would at least enable a missing value to be treated the same as a present value that is set to False.

Reply:
updated to make `skip_hdus` a list/tuple/set of extnames to skip; if an extension isn't in the list then it isn't skipped.